src/reference/asciidoc/redis.adoc

```diff
@@ -769,6 +769,94 @@ IMPORTANT: The `task-executor` has to be configured with more than one thread fo
+It defaults to `true`.
+====
+
+[[redis-stream-inbound]]
+=== Redis Stream Inbound Channel Adapter
+
+Spring Integration 5.4 introduced the Reactive Stream inbound channel adapter for reading messages from a Redis Stream.
+Inbound channel adapter uses `StreamReceiver.receive(...)` or `StreamReceiver.receiveAutoAck` based on auto acknowledgement flag to read `Message` from redis stream.
+The following example shows how to use JavaConfiguration for Redis Stream Inbound Channel Adapter.
+
+====
+[source,java]
+----
+@Component
+public class MyClass {
+
+    @Bean
+    public ReactiveRedisStreamMessageProducer reactiveRedisStreamProducer(ReactiveRedisConnectionFactory reactiveRedisConnectionFactory) {
+    ReactiveRedisStreamMessageProducer messageProducer =
+                new ReactiveRedisStreamMessageProducer(reactiveRedisConnectionFactory, "myStreamKey"); <1>
+        messageProducer.setStreamReceiverOptions( <2>
+                    StreamReceiver.StreamReceiverOptions.builder()
+                          .pollTimeout(Duration.ofMillis(100))
+                          .build());
+        messageProducer.setAutoStartup(true); <3>
+        messageProducer.setAutoAck(false); <4>
+        messageProducer.setCreateConsumerGroup(true); <5>
+        messageProducer.setConsumerGroup("my-group"); <6>
+        messageProducer.setConsumerName("my-consumer"); <7>
+        messageProducer.setOutputChannel(""); <8>
+        messageProducer.setReadOffset(ReadOffset.latest()); <9>
+        messageProducer.extractPayload(true); <10>
+        return messageProducer;
+    }
+}
+----
+<1> Construct an instance of `ReactiveRedisStreamMessageProducer` using `ReactiveRedisConnectionFactory` and `StreamKey`.
+<2> `StreamReceiver.StreamReceiverOptions` Options to consume redis stream using reactive infrastructure.
+<3> A `SmartLifecycle` attribute to specify whether this endpoint should start automatically after the application context start or not.
+It defaults to `true`.
+If `false`, `RedisStreamMessageProducer` should be started manually `messageProducer.start()`.
+<4> If `false`, Received messages are not auto acknowledged. The acknowledgement of the message will be deferred to the client consuming message.
+It defaults to `true`.
+
+    If there is `SerializationException` message will be sent to `error-channel` and auto-acknowledged.
+
+<5> If `true`, Consumer Group will be created. During creation of consumer group stream will be created (if not exists yet), too.
+Consumer group track message delivery and distinguish between consumers.
+It defaults to `false`.
+<6> Set Consumer Group name.
+It defaults to the defined bean name.
+<7> Set Consumer name.
+Read message as `my-consumer` from group `my-group`.
+<8> The message channel to which to send `Message` instances from this endpoint.
+<9> Define the offset to read message.
+It defaults to `ReadOffset.latest()`.
+<10> If 'true', channel adapter will extract payload from `Message`
```

Comment:
That's something misleading. What does it extract and from what?
The point of this inbound endpoint is really read from Redis, build a message and produce it.
See the behavior around that `extractPayload` in the `ReactiveRedisStreamMessageProducer`